multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -24,20 +31,672 @@ use node_lib::database::db_initializer::{
+            // This balance is of a middle size it will be reduced as there won't be enough
+            // after the first one is filled up.
+            owed_to_serving_node_2: AccountedDebt {
+                balance_minor: owed_to_serv_node_2_minor,
+                age_s: payment_thresholds.maturity_threshold_sec + 100_000,
+            },
+            // This account will be the least significant and therefore eliminated
+            owed_to_serving_node_3: AccountedDebt {
+                balance_minor: owed_to_serv_node_3_minor,
+                age_s: payment_thresholds.maturity_threshold_sec + 30_000,
+            },
+        }),
+        payment_thresholds_all_nodes: payment_thresholds,
+        cons_node_transaction_fee_agreed_unit_price_opt: Some(
+            agreed_transaction_fee_unit_price_major,
+        ),
+        // It seems like the ganache server sucked up quite less than those 55_000 units of gas??
+        exp_final_cons_node_transaction_fee_balance_minor: 2_828_352_000_000_001,
+        // Zero reached, because the algorithm is designed to exhaust the wallet completely
+        exp_final_cons_node_service_fee_balance_minor: 0,
+        // This account was granted with the full size as its lowest balance from the set makes
+        // it weight the most
+        exp_final_service_fee_balance_serv_node_1_minor: owed_to_serv_node_1_minor,
+        exp_final_service_fee_balance_serv_node_2_minor: owed_to_serv_node_2_minor
+            - gwei_to_wei::<u128, u64>(2_345_678),
+        // This account dropped out from the payment, so received no money
+        exp_final_service_fee_balance_serv_node_3_minor: 0,
+    };
+
+    let process_scan_request_to_node =
+        |real_node: &MASQRealNode, ui_port: u16, scan_type: ScanType, context_id: u64| {
+            let ui_client = real_node.make_ui(ui_port);
+            ui_client.send_request(UiScanRequest { scan_type }.tmb(context_id));
+            let response = ui_client.wait_for_response(context_id, Duration::from_secs(10));
+            UiScanResponse::fmb(response).unwrap();
+        };
+
+    let stimulate_payments =
+        |_cluster: &mut MASQNodeCluster,
+         real_consuming_node: &MASQRealNode,
+         _global_test_config: &TestInputsOutputsConfig| {
+            process_scan_request_to_node(
+                &real_consuming_node,
+                consuming_node_ui_port,
+                ScanType::Payables,
+                1111,
+            )
+        };
+
+    let start_serving_nodes_and_run_check = |cluster: &mut MASQNodeCluster,
+                                             serving_node_1_attributes: ServingNodeAttributes,
+                                             serving_node_2_attributes: ServingNodeAttributes,
+                                             serving_node_3_attributes: ServingNodeAttributes,
+                                             test_global_config: &TestInputsOutputsConfig|
+     -> (MASQRealNode, MASQRealNode, MASQRealNode) {
```

Comment:
Optional: If the deconstruction of returned value is not essential, then maybe you'd limit the length using an array.

```rust
-> [MASQRealNode; 3]
```

```rust
let iter = 1..3;
let (x, y) = iter.collect_tuple().unwrap(); // yeah!
```